### PR TITLE
Added Sync API for  public interface methods.

### PR DESCRIPTION
**Pull request by yashlad681**
### Summary
Added @sync_compatible decorator to the functions read_path,write_path,get_directory,put_directory in all classes of filesystems.py for the following PR https://github.com/PrefectHQ/prefect/issues/645

### Checklist

This pull request is:

- [x] A short code fix

### CODE REVIEW

**Comment by madkinsz** [comment]
Nice work! It's failing a formatting check, see https://docs.prefect.io/contributing/overview/#setting-up-a-development-environment for details on setting up pre-commit hooks.

**Comment by yashlad681** [comment]
> Nice work! It's failing a formatting check, see https://docs.prefect.io/contributing/overview/#setting-up-a-development-environment for details on setting up pre-commit hooks.

Hi, I followed each and every step from that link you shared but I dont know why it fails for black as pre-commit hook.

**Comment by yashlad681** [comment]
> Nice work! It's failing a formatting check, see https://docs.prefect.io/contributing/overview/#setting-up-a-development-environment for details on setting up pre-commit hooks.

Thanks, pre-commit wasn't as easy as it looked like. 😅

**Comment by yashlad681** [comment]
Hey @madkinsz can you review this and let  me know in case if there are any gaps ?

Thanks

**Comment by madkinsz** [comment]
This looks good! Can you add some test for calling the methods from a function that isn't async in https://github.com/PrefectHQ/prefect/blob/main/tests/test_filesystems.py#L8 ? e.g. `def test_read_write_roundtrip_sync`

**Comment by yashlad681** [comment]
Hi @madkinsz actually tried learning from different sources how to write unit tests but having hard time 
But once an engineer writes I would like to learn how to write them
Apologies for not informing earlier.

**Comment by robfreedy** [comment]
@Bada-S is taking a look at writing these tests over the next couple of weeks, will keep y'all posted as he makes progress!

**Comment by Bada-S** [comment]
I added tests for sync functions and ran pre-commit to fix formatting issues

**Comment by Bada-S** [comment]
@yashlad681 Hi, I made some changes to your pr. Firstly, I installed pre-commit and the pre-commit hooks reformatted the files. Looks like that fixed some of the python tests. Secondly, I added some tests for the methods with the '@sync_compatible` decorator to test them in the case that they are called from sync functions

File: tests/test_filesystems.py [hunk 1]
```diff
@@ -56,26 +78,49 @@ def test_must_contain_net_location(self):
...
+
     async def test_write_outside_of_basepath_netloc(self):
         fs = RemoteFileSystem(basepath="memory://foo")
         with pytest.raises(ValueError, match="is outside of the base path"):
             await fs.write_path("memory://bar/test.txt", content=b"hello")
 
+    def test_write_outside_of_basepath_netloc_sync(self):
+        fs = RemoteFileSystem(basepath="memory://foo")
+        with pytest.raises(ValueError, match="is outside of the base path"):
+            fs.write_path("memory://bar/test.txt", content=b"hello")
```
Comment (madkinsz) [inline tests/test_filesystems.py:112]: We actually don't want to cover every possible test case with `sync`. They go through the same code and duplicated coverage will be more work to maintain. We just want to make sure the method is usable, the `roundtrip` test should be all we need. Would you mind removing the extras?

Comment (Bada-S) [reply tests/test_filesystems.py]: I removed all of those sync tests except the roundtrip sync ones

**Review by madkinsz — APPROVED** [review]

**Comment by yashlad681** [comment]
> @yashlad681 Hi, I made some changes to your pr. Firstly, I installed pre-commit and the pre-commit hooks reformatted the files. Looks like that fixed some of the python tests. Secondly, I added some tests for the methods with the '@sync_compatible` decorator to test them in the case that they are called from sync functions. If you need me to explain any of those changes, please let me know!

Hi @Bada-S  , Am going through those tests trying to understand. thanks

**Pull request merged**